crates/service_utils/src/helpers.rs

```diff
@@ -401,3 +411,91 @@ pub fn parse_config_tags(
         }
     }
 }
+
+pub async fn execute_webhook_call<T>(
+    webhook_object: Value,
+    payload: T,
+    config_version_opt: &Option<String>,
+    event: WebhookEvent,
+) -> Result<(), AppError>
+where
+    T: Serialize,
+{
+    match serde_json::from_value::<Webhook>(webhook_object) {
+        Ok(webhook) => {
+            let http_client = reqwest::Client::new();
+            let mut header_array = webhook
+                .headers
+                .into_iter()
+                .filter_map(|key| match key {
+                    HeadersEnum::ConfigVersion => {
+                        if let Some(config_version) = config_version_opt {
+                            Some((
+                                "x-config-version".to_string(),
+                                config_version.to_string(),
+                            ))
+                        } else {
+                            None
+                        }
+                    }
+                })
+                .collect::<Vec<(String, String)>>();
+
+            if let Ok(auth) =
+                serde_json::from_value::<Authorization>(webhook.authorization)
+            {
+                let auth_token_value: String =
+                    get_from_env_unsafe(&auth.value).map_err(|err| {
+                        log::error!("Failed to retrieve authentication token for the webhook with error: {}", err);
+                        AppError::WebhookError(
+                            anyhow!("Failed to retrieve authentication token for the webhook. Please verify the credentials in TenantConfig.")
+                        )
+                    })?;
+                header_array.push((auth.key, auth_token_value));
+            }
+
+            let mut headers = HeaderMap::new();
+            header_array.iter().for_each(|(name, value)| {
+                let h_name = HeaderName::from_str(name);
+                let h_value = HeaderValue::from_str(value);
+
+                if let (Ok(key), Ok(value)) = (h_name, h_value) {
+                    headers.insert(key, value);
+                }
+            });
+
+            let request_builder = match webhook.method {
+                HttpMethod::Post => http_client.post(&webhook.url),
+                HttpMethod::Put => http_client.put(&webhook.url),
+                HttpMethod::Get => http_client.get(&webhook.url),
+            };
+
+            let response = request_builder
+                .headers(headers.into())
+                .json(&WebhookResponse { event, payload })
+                .send()
+                .await;
+
+            match response {
+                Ok(res) => match res.status() {
+                    StatusCode::OK => {
+                        log::info!("webhook call succeeded: {:?}", res);
```

Comment:
```suggestion
                        log::info!("webhook call succeeded: {}", res.to_string());
```

Reply:
We maybe shouldn't log the response, the service we called could return a huge file or response and break us. We can instead log some details - like status code, request headers, response headers. We don't care about the body of the response